app/components/search.tsx

```diff
@@ -378,3 +379,41 @@ const normalize = (question: string) =>
  * Sort function for the highest score on top
  */
 const byScore = (a: SearchResult, b: SearchResult) => b.score - a.score
+
+/**
+ * Handle flushing searched phrases to the NLP logger endpoint
+ */
+let prevSearch = ''
+let lastTimestamp = 0
+
+const logSearch = (value: string) => {
+  setTimeout(shouldFlushSearch(value, prevSearch), 4000)
```

Comment:
this is a potential heisenbug, as it's possible for the flush calls to come in the wrong order? Should be fine?

Reply:
The search debounce is 100ms, which is not much. I'd prefer not to have logs full of things like the following:
```
Wh
Why
Why d
Why do
Why do I
Why do I ty
Why do I typ
```
This is good when searching, as the results should update often, but in the case of logging searches, it seems overkill - better to just log things which are relatively complete (whatever that means :D).

Debouncing will help with avoiding the multiple entries, at the cost of filtering out people who rapidly write in several queries before finding what they're looking for. I don't know what would be best

Reply:
> better to just log things which are relatively complete

Is it possible to only log searches when the user clicks outside of search bar, i.e. selects something from dropdown menu or just abandons search? It'd be interesting to log the query as well as selection or abandonment if possible.

Reply:
hmm, 100ms is quite short ... when/why did we change it from 300 / 500ms like we had before?

in any case, I still vote that whatever is good for the search feature should also be good for logging - and if not, we need tests for the logging correctness

Reply:
Updated late Feb because people were complaining about lag:
Why - https://github.com/StampyAI/stampy-ui/issues/175
When - https://github.com/StampyAI/stampy-ui/pull/186/files/7ffdf8becbf424eb155ef198d0c518b5ac4ea354